File: java/src/main/java/co/edu/sena/ejercicios/ejercicio12/Cliente.java
package co.edu.sena.ejercicios.ejercicio12;

import java.util.List;

public class Cliente {
    private String tipoDocumento;
    private String numeroDocumento;
    private String nombre;
    private String direccion;
    private String telefono;
    private List<Factura> facturas;

    public String getTipoDocumento() {
        return tipoDocumento;
    }

    public void setTipoDocumento(String tipoDocumento) {
        this.tipoDocumento = tipoDocumento;
    }

    public String getNumeroDocumento() {
        return numeroDocumento;
    }

    public void setNumeroDocumento(String numeroDocumento) {
        this.numeroDocumento = numeroDocumento;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public List<Factura> getFacturas() {
        return facturas;
    }

    public void setFacturas(List<Factura> facturas) {
        this.facturas = facturas;
    }



    @Override
    public String toString() {
        return "Cliente{" +
                "tipoDocumento='" + tipoDocumento + '\'' +
                ", numeroDocumento='" + numeroDocumento + '\'' +
                ", nombre='" + nombre + '\'' +
                ", direccion='" + direccion + '\'' +
                ", telefono='" + telefono + '\'' +
                ", facturas=" + facturas +
                '}';
    }
}

File: typescript/src/ejercicio12/cliente.ts
import {Factura} from "./factura";

export class Cliente {
    private _tipoDocumento: string;
    private _numeroDocumento: string;
    private _nombre: string;
    private _direccion: string;
    private _telefono: string;
    private _facturas: Factura[];


    get tipoDocumento(): string {
        return this._tipoDocumento;
    }

    set tipoDocumento(value: string) {
        this._tipoDocumento = value;
    }

    get numeroDocumento(): string {
        return this._numeroDocumento;
    }

    set numeroDocumento(value: string) {
        this._numeroDocumento = value;
    }

    get nombre(): string {
        return this._nombre;
    }

    set nombre(value: string) {
        this._nombre = value;
    }

    get direccion(): string {
        return this._direccion;
    }

    set direccion(value: string) {
        this._direccion = value;
    }

    get telefono(): string {
        return this._telefono;
    }

    set telefono(value: string) {
        this._telefono = value;
    }

    get facturas(): Factura[] {
        return this._facturas;
    }

    set facturas(value: Factura[]) {
        this._facturas = value;
    }

}